pennylane/estimator/templates/qubitize.py

```diff
@@ -343,20 +394,19 @@ def controlled_resource_decomp(
             )
         )
 
-        if not prep_op:
-            coeff_precision = target_resource_params["coeff_precision"]
+        prep_kwargs = {"thc_ham": thc_ham}
+        if coeff_precision:
+            prep_kwargs["coeff_precision"] = coeff_precision
+
+        if coeff_precision or prep_op is None:
             # Prep cost from Fig 3 and 4 in arXiv:2011.03494
-            prep_op = resource_rep(
-                PrepTHC,
-                {"thc_ham": thc_ham, "coeff_precision": coeff_precision},
-            )
+            prep_op = resource_rep(PrepTHC, prep_kwargs)
+
```

Comment:
Similar comments as above. The coeff precision should be well defined so we can simplify these blocks.